content/en/blog/_posts/2024-08-19-pod-failure-policy-ga.md

```diff
@@ -0,0 +1,134 @@
+  or for a custom Pod condition that you add using mechanisms like webhooks or external controllers.
+
+For each rule, you can specify one of the following actions to take when a Pod
+matches the specified match condition:
+- `Ignore` - don't count the failure towards the `.backoffLimit`
+- `FailJob` - fail the entire Job and terminate all running Pods
+- `FailIndex` - in an indexed Job that specifies a
+  [Backoff limit per index](/docs/concepts/workloads/controllers/job/#backoff-limit-per-index),
+  fail the index that corresponds to the failed Pod
+- `Count` - count the failure towards the `backoffLimit` or `backoffLimitPerIndex`.
+  This is the default behavior.
+
+When you run your Job and Pod failures occur, the Job controller matches the
+failed Pod status against the list of Pod failure policy rules, in order, and
+takes the corresponding actions for each match.
+
+### Example
+
+The Pod failure policy snippet below demonstrates an example use:
+
+```yaml
+podFailurePolicy:
+  rules:
+  - action: Ignore
+    onPodConditions:
+    - type: DisruptionTarget
+  - action: FailJob
+    onPodConditions:
+    - type: ConfigIssue
+  - action: FailJob
+    onExitCodes:
+      operator: In
+      values: [ 42 ]
+```
+
+This example, the Pod failure policy does the following:
+
+- ignores any failed Pods that have the built-in `DisruptionTarget`
+  condition. These Pods don't count towards Job backoff limits.
+- fails the Job if any failed Pods have the custom user-supplied
+  `ConfigIssue` condition, which was added either by a custom controller or webhook.
+- fails the Job if any containers exited with the reserved exit code 42
+- counts all other Pod failures towards the default `backoffLimit` (or
+  `backoffLimitPerIndex` if used).
+
+## Related work
+
+We are pleased to observe the follow up work based on the concepts introduced
+in this work, comprising:
+- [Add more granular failure reason for Job PodFailurePolicy](https://github.com/kubernetes/enhancements/issues/4443)
+- JobSet [Feature: Configurable Failure Policy API](https://github.com/kubernetes-sigs/jobset/issues/262)
+- Proposal [Disrupted Pods should be removed from endpoints](https://docs.google.com/document/d/1t25jgO_-LRHhjRXf4KJ5xY_t8BZYdapv7MDAxVGY6R8)
+- Support for Pod Failure Policy via JobSet in [Kubeflow Training v2](https://github.com/kubeflow/training-operator/pull/2171)
+
+## Learn more
+
+- Read the user-facing documentation for
+  [Pod failure policy](/docs/concepts/workloads/controllers/job/#pod-failure-policy) and
+  [Backoff limit per index](/docs/concepts/workloads/controllers/job/#backoff-limit-per-index)
+- Read the tasks page on
+  [Handling retriable and non-retriable pod failures with Pod failure policy](https://kubernetes.io/docs/tasks/job/pod-failure-policy/)
+- Read the KEP for [Pod failure policy](https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/3329-retriable-and-non-retriable-failures)
+
```

Comment:
Additionally, it might be better to mention the Pod Disruption Conditions documentation: https://kubernetes.io/docs/concepts/workloads/pods/disruptions/#pod-disruption-conditions

WDYT?

Reply:
Good idea, added